controller/tests/update_tests.ml

```diff
@@ -0,0 +1,99 @@
+open Update
+open Lwt
+open Update_test_helpers
+
+let happy_flow_test () =
+  let init_state = GettingVersionInfo in
+  let current_version = "10.0.1" in
+  let next_version = "10.0.2" in
+
+  let expected_bundle_name vsn =
+      Mock_update_client.test_bundle_name ^ _MAGIC_PAT ^ vsn ^ _MAGIC_PAT
+  in
+  let expected_url =
+    Config.System.update_url ^ _MAGIC_PAT
+        ^ expected_bundle_name next_version
+  in
+
+  let expected_state_sequence =
+    [
+      UpdateMock (fun () ->
+        Mock_rauc.set_status SystemA {
+            Mock_rauc.some_status with version = current_version
+        };
+        Mock_update_client.add_bundle next_version
+            ("BUNDLE_CONTENTS: " ^ next_version);
+        Mock_update_client.set_latest_version next_version;
+      );
+      StateReached GettingVersionInfo;
+      StateReached (Downloading { url = expected_url; version = next_version });
+      StateReached (Installing (_MAGIC_PAT ^ expected_bundle_name next_version));
+      ActionDone
+        ( "bundle was installed and marked as primary",
+          fun () ->
```

Comment:
Yes, absolutely, that was the original idea - to parameterize the `fun`'s over the test state, but I just didn't get to it. Will attempt to refactor it in this fashion.

Reply:
I pushed an update that resolves this, but I parameterizing over the whole scenario spec, rather than individual scenario specs, because the scenario always uses the same mocks anyway, so no need to pass them each time.